### PR TITLE
Use tangle runtime metadata

### LINKED ISSUE #460: [TASK] Use tangle runtime rust bindings in relayer
## Overview
Currently, the relayer uses `substrate-runtime` and `dkg-runtime`  [metadata](https://docs.rs/webb/0.5.21/webb/substrate/index.html) separately to interact with respective nodes. But since the relayer will be interacting with just the `tangle` node, we should update the relayer code base to use rust bindings for tangle-runtime metadata.

We can also remove substrate-runtime and dkg-runtime from webb-rs.

### LINKED ISSUE #481: [TASK] Add tangle-substrate-types
## Overview
Now that we are upgrading the relayer to use the tangle node, we should generate types for the tangle node and use them in relayer tests.

## Checklist
- [x] Deprecate `protocol-substrate-types` package
- [x] Deprecate `dkg-substrate-types` package
- [x] Deprecate `substrate-runtime` from webb-rs 
- [x] Deprecate `dkg-runtime` from webb-rs
- [x] Add `tangle-substrate-types` package https://github.com/webb-tools/tangle/pull/167
- [x] Update Relayer with new types

### PR DESCRIPTION
**Pull request by salman01zp**
## Summary of changes
- [x] Update the relayer to use tangle-metadata to interact with the tangle node
- [x] Refactor relayer configuration(remove runtime-type)
- [x] Update tests to use tangle node for testing
- [x]  Create tangle docker image for integration testing

---
- Depends on https://github.com/webb-tools/tangle/pull/160
- Depends on https://github.com/webb-tools/tangle/pull/161
- Depends on https://github.com/webb-tools/tangle/pull/164
- Depends on https://github.com/webb-tools/tangle/issues/165

### Reference issue to close (if applicable)
- Closes #460 
- Closes #481 

-----
### Code Checklist 

- [ ] Tested
- [ ] Documented

### CODE REVIEW

**Comment by shekohex** [comment]
Please link the issue/task related to this PR

**Comment by shekohex** [comment]
Any updates on this @salman01zp ?

**Comment by salman01zp** [comment]
Not yet, this will require cleaning relayer tests and just having one localTangle.ts file interact with the tangle node. Will continue once the relayer is upgraded to node v18

**Comment by salman01zp** [comment]
The tangle docker image is not producing blocks. Trying with local setup

**Comment by salman01zp** [comment]
I tried with docker local setup using Mac(arm64) and it works.  But for some reason, it fails for the docker image published by CI (x86_64).
I have opened issue here

**Comment by salman01zp** [comment]
substrate vanchor private transaction tests is failing with `Invalid transaction proof`. I have skipped that test for now.
I will open a follow-up PR to fix that test after debugging

File: tests/test/substrate/vanchorPrivateTransaction.test.ts [hunk 1]
```diff
@@ -46,17 +44,18 @@ import {
...
+import { currencyToUnitI128, UsageMode } from '@webb-tools/test-utils';
 import {
+  createAccount,
   defaultEventsWatcherValue,
   generateVAnchorNote,
 } from '../../lib/utils.js';
+import { LocalTangle } from '../../lib/localTangle.js';
 
-describe('Substrate VAnchor Private Transaction Relayer Tests', function () {
+describe.skip('Substrate VAnchor Private Transaction Relayer Tests', function () {
```
Comment (shekohex) [inline tests/test/substrate/vanchorPrivateTransaction.test.ts:55]: A PR to make this test work is needed.

**Review by shekohex — APPROVED** [review]
Overall this looks good!

**Pull request merged**